06_puppy/jimbotech/mapStore_test.go

```diff
@@ -0,0 +1,131 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type storesSuite struct {
+	suite.Suite
+	store  Storer
+	mapper mapCheck
+}
+
+func TestStorer(t *testing.T) {
+	m := NewPuppyStorer()
+	ms := &SyncMapStore{}
+	suite.Run(t, &storesSuite{store: m, mapper: m})
+	suite.Run(t, &storesSuite{store: ms, mapper: ms})
+}
+
+// TestMapStoreWithoutContructor may not be required as it may not be
+// possible to create an instance of that type without using the constructor
+func TestMapStoreWithoutContructor(t *testing.T) {
+	var puppyStore MapStore
+	pup := Puppy{1, "kelpie", "brown", "indispensable"}
+
+	_, err := puppyStore.CreatePuppy(&pup)
+	f := func(t *testing.T) {
+		assert.NotNil(t, err)
+		assert.Equal(t, "store not created, call initializing constructor first", err.Error())
+	}
+	t.Run("CreatePuppy test", f)
+
+	err = puppyStore.UpdatePuppy(1, &pup)
+	t.Run("UpdatePuppy test", f)
+	_, err = puppyStore.ReadPuppy(1)
+	t.Run("ReadPuppy test", f)
+	puppyStore.DeletePuppy(1)
+}
+
+// TestCreateSuccess add to the store and verify
+// by reading that it is in the store and then remove it
+// to leave the store in the original state
+func (s *storesSuite) TestCreateSuccess() {
+	asserter := assert.New(s.T())
+	if success, pup := create(s); success {
+		// now check by reading the value back and compare
+		pup2, err2 := s.store.ReadPuppy(pup.ID)
+		if asserter.Nil(err2, "Read store should work") {
+			asserter.Equal(*pup, *pup2)
+		}
+		s.store.DeletePuppy(pup.ID)
+	}
+}
+
+func create(s *storesSuite) (bool, *Puppy) {
+	asserter := assert.New(s.T())
+	pup := Puppy{1, "kelpie", "brown", "indispensable"}
+	id, err := s.store.CreatePuppy(&pup)
+	if asserter.Nil(err, "Create on the store should have worked") {
```

Comment:
If you use `require.NoError(err)` here, you don't need the nesting, and you can get rid of the bool return value.

Reply:
Done